apps/prairielearn/python/colors.py

```diff
@@ -190,7 +47,10 @@ def get_css_color(name: str) -> Optional[str]:
     """
 
     name = name.lower()
-    if name in colors:
-        return colors[name]
-    else:
+    if name in pl_colors:
+        return pl_colors[name]
+    try:
+        return Color(name).to_string(hex=True)
+    except ValueError:
```

Comment:
Is there a way to do this without exception handling? Like can we just grab the list of named colors from coloraide somehow?

Reply:
I don't think so, just taking a look at [the constructor](https://github.com/facelessuser/coloraide/blob/4bb07fd09e0fb3010706bcf549464dadae977ee6/coloraide/color.py#L243-245). The `cls._match(...)` it's using isn't exposed as far as I can tell.

Reply:
```
>>> from coloraide import Color
>>> Color.match('red')
ColorMatch(color=color(srgb 1 0 0 / 1), start=0, end=3)
>>> Color.match('invalid_color')
>>> Color.match('invalid_color') is None
True
```

https://facelessuser.github.io/coloraide/color/?h=match#color-matching

Reply:
Not sure how I missed that... thank you! Will change this over.